### PR TITLE
Cert manager v1.3 - Approval Condition

### PR DESCRIPTION
**Pull request by JoshVanL**
In cert-manager v1.3, we introduced the [approval mechanism](https://cert-manager.io/docs/concepts/certificaterequest/#approval). This PR adds an optional check, which is enabled by default, to wait for the presence of this Approval condition before signing.

This PR also adds RBAC which enables the internal cert-manager approver to approve all KMS CertificateRequests.

Once this is merged and released we can move this issuer to the list of approval honoured issuers [here](https://cert-manager.io/docs/configuration/external/#issuers-that-honour-approval)

/assign @skygeo

### CODE REVIEW

**Comment by maruina** [comment]
Hey @JoshVanL, thank you for your PR. I'll go over it in the next couple of days.

I need to figure out what's wrong with the CI. We should switch to Github Actions like our other controllers.

File: config/crd/bases/cert-manager.skyscanner.net_kmsissuers.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.2.5
+    controller-gen.kubebuilder.io/version: (unknown)
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply config/crd/bases/cert-manager.skyscanner.net_kmsissuers.yaml]: Thanks @maruina, I have changed controller-gen to a proper version now.

File: controllers/certificaterequest_controller.go [hunk 2]
```diff
@@ -161,6 +174,64 @@ func (r *CertificateRequestReconciler) SetupWithManager(mgr ctrl.Manager) error
 // 	return false
 // }
 
+// requestShouldBeProcessed will return false if the conditions on the request
+// mean that it should not be processed. If the request has been denied, it
+// will set the request failure time and add a Ready=False condition.
+func (r *CertificateRequestReconciler) requestShouldBeProcessed(ctx context.Context, log logr.Logger, cr *cmapi.CertificateRequest) (bool, error) {
```
Comment (maruina) [inline controllers/certificaterequest_controller.go:180]: Are you planning to add a test for this function or do you think it is not necessary?

**Review by maruina — COMMENTED** [review]
Hey @JoshVanL thanks for your PR!
I added a couple of comments, but in general LGTM

File: main.go [hunk 3]
```diff
@@ -86,6 +91,9 @@ func main() {
 		Log:      ctrl.Log.WithName("controllers").WithName("CertificateRequest"),
 		Recorder: mgr.GetEventRecorderFor("certificaterequests-controller"),
 		KMSCA:    ca,
+
+		Clock:                  clock.RealClock{},
+		CheckApprovedCondition: !disableApprovedCheck,
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply main.go:96]: Yep, happy to change it this way!

File: controllers/certificaterequest_controller.go (hunk 2 above)
Comment (JoshVanL) [reply controllers/certificaterequest_controller.go:180]: Wasn't planing to but happy to add some tests in.

File: main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (maruina) [reply main.go:96]: Cheers, thanks.

File: controllers/certificaterequest_controller.go (hunk 2 above)
Comment (maruina) [reply controllers/certificaterequest_controller.go:180]: It would be great, thanks.

**Review by maruina — APPROVED** [review]
Thanks for your contribution! LGTM

**Pull request merged**